src/main/java/storage/Storage.java

```diff
@@ -139,6 +133,7 @@ public AuthorList readAuthorListFromDataFile() {
      */
     public void saveAuthorListToDataFile(AuthorList authorList) {
         assert authorList != null : "authorList cannot be null";
+        createFileIfNeeded();
```

Comment:
Seems like if `createFileIfNeeded` fails, the rest of the method still continues which shouldn't be the case since the actual file/file location could not be created.

Reply:
Good catch! Will fix soon.